src/config/from_document.rs

```diff
@@ -75,15 +78,66 @@ fn process_schema_directives<T: DirectiveCodec<T> + Default>(
     res
 }
 
+fn process_schema_multiple_directives<T: DirectiveCodec<T> + Default>(
+    schema_definition: &SchemaDefinition,
+    directive_name: &str,
+) -> Valid<Vec<T>, String> {
+    let directives: Vec<Valid<T, String>> = schema_definition
+        .directives
+        .iter()
+        .filter_map(|directive| {
+            if directive.node.name.node.as_ref() == directive_name {
+                Some(T::from_directive(&directive.node))
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    let mut res = Valid::succeed(Vec::new());
+
+    for directive in directives {
+        res = res.zip(directive).map(|(mut vec, item)| {
+            vec.push(item);
+            vec
+        });
+    }
+
+    res
+}
+
 fn server(schema_definition: &SchemaDefinition) -> Valid<Server, String> {
     process_schema_directives(schema_definition, config::Server::directive_name().as_str())
 }
+
 fn upstream(schema_definition: &SchemaDefinition) -> Valid<Upstream, String> {
     process_schema_directives(
         schema_definition,
         config::Upstream::directive_name().as_str(),
     )
 }
+
+fn links(schema_definition: &SchemaDefinition) -> Valid<Vec<Link>, String> {
+    let links: Valid<Vec<Link>, String> = process_schema_multiple_directives(
+        schema_definition,
+        config::Link::directive_name().as_str(),
+    );
+
+    links.and_then(|links| {
+        let mut ids: Vec<String> = Vec::new();
+        for link in &links {
+            if let Some(id) = &link.id {
+                if ids.contains(id) {
+                    return Valid::fail(format!("Duplicated id: {}", id))
+                        .trace(config::Link::directive_name().as_str());
+                }
+                ids.push(id.to_string());
+            }
+        }
+        Valid::succeed(links)
```

Comment:
This logic will bail out on the first duplicate found. 
IMPORTANT - Write a test case to check if multiple duplicates are reported.

Reply:
[tests/graphql/errors/test-link.graphql](https://github.com/tailcallhq/tailcall/pull/1094/files#diff-11d605567b472743fd4e0d46c87318f6f1ca28a6eff75d0307f26164ce8510d7) covers this, I'll change its name to make clearer

Reply:
test renamed